### PR TITLE
empty branch for full code review

### PR DESCRIPTION
**Pull request by dwoznicka**

### CODE REVIEW

File: app/index.html [hunk 1]
```diff
@@ -0,0 +1,316 @@
...
+                </section>
+              </div>
+            </div>
+          </div>
+        </div>
+      </div>
+      <div class="col-sm-12 col-md-12 col-lg-4">
+        <div class="row">
+          <div class="col-sm-12 col-md-12 col-lg-12">
+            <canvas id="categoriesChart" height="400"></canvas>
```
Comment (dwoznicka) [inline app/index.html]: please stick to one convention of id syntax (just as classnames) either it is camelCase or hyphen-separated, or snake_case, every id in the project should follow one convention :)

File: app/index.html [hunk 2]
```diff
@@ -0,0 +1,316 @@
...
+          </div>
+        </div>
+        <div class="row">
+          <div class="col-sm-12 col-md-12 col-lg-12">
+            <canvas id="placesChart" height="400"></canvas>
+          </div>
+        </div>
+        <div class="row">
+          <div class="col-sm-12 col-md-12 col-lg-12">
+            <div id="add-new-place" class="new-place-button" data-toggle="modal" data-target="#newPlaceModal">
```
Comment (dwoznicka) [inline app/index.html]: this doesn't seem like a good usage of maintainablecss. `new` doesnt say anything about module, `newPlace-button` is way better. besides, we should definetely practice syntactic HTML , meaning we should use tag <button> when we want to achieve a button-like div, and use either <button> or <a> for any interactive tag (clickable)

File: app/resources/css/main.css [hunk 3]
```diff
@@ -0,0 +1,212 @@
+body {
+  color: #ffffff;
+  font-family: 'Barlow Condensed', sans-serif;
```
Comment (dwoznicka) [inline app/resources/css/main.css]: i want to encourage you here to research about FOUT and FOIT phenomenons :)

when you apply a font family like this, an ugly roman font is first displayed, then it disappears when browser notices the custom one is downloaded, and before displaying the new one we have a flash of invisible text.
we can minimalize this effect by following this tutorial
https://www.filamentgroup.com/lab/font-events.html

**tldr;** in short, what you want to do is to create a variable for your custom font family, let's say 
`$font-base-family: 'Barlow Condensed', sans-serif;`

and here apply only 'sans-serif'.
then you download Bram Stein’s FontFaceObserver script [https://github.com/bramstein/fontfaceobserver] (just the `fontfaceobserver.js`, but read the readme there - the standalone version is lighter, but we prefer not to use it) and paste it into the project

in a js file, preferably somewhere at the top (one of the first js files that will be read) you place a a self-calling script 
```
var font = new FontFaceObserver('Barlow Condensed');

font.load().then(function () {
  document.documentElement.className += " fonts-loaded";
});
```
so it's called before document.ready. 
then you must add another style so your font is applied *only after it is fully loaded* - 
```
.fonts-loaded {
  body {
    font-family: 'Barlow Condesed', sans-serif;
  }
}
```

you can later play with cookies etc so the page loads even faster, and is readable muuuuch earlier. 
it is especially desired on blogs and articles, where user probably starts reading right away, and any font/line jumping is extremely disturbing.
there's much more info about it in the atttached tutorial :)

File: app/resources/css/main.css [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+.content-tabs {
+  border: none; }
+
+.content-tabs-item {
+  border: none;
+  margin-right: 2px;
+  border-top-right-radius: 5px;
+  border-top-left-radius: 5px; }
+
+.nav-tabs > li > a {
```
Comment (dwoznicka) [inline app/resources/css/main.css]: please avoid selecting by tags. we have maintainablecss not to have to write what you have here. if you applied an unique classname to that element (<a> tag), you would only style that classname here

File: app/resources/sass/layout/header.scss [hunk 5]
```diff
@@ -0,0 +1,22 @@
+#login-register,
```
Comment (dwoznicka) [inline app/resources/sass/layout/header.scss]: please avoid styling by id as a selector. they should serve for JS only.

Comment (dwoznicka) [reply app/resources/sass/layout/header.scss]: those elements can have the same class - it will be less code and less headache :)

File: app/resources/sass/components/placeCard.scss [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+  color: $white !important;
+  transition: 0.5s;
+}
+
+.content-place-cards {
+  border-top: 1px solid $orange-main;
+  margin-bottom: 30px;
+}
+
+.br-theme-fontawesome-stars-o .br-widget a::after {
```
Comment (dwoznicka) [inline app/resources/sass/components/placeCard.scss]: by using proper maintainablecss classnames you can avoid such long selectors ;) just think well of the classname so in one string you can now exactly which link you are stuyling. eg. `.placeCard-starsWidget-link`

File: app/resources/js/api-client.js [hunk 7]
```diff
@@ -0,0 +1,163 @@
...
+  const API_URL = "http://taste-api.binarlab.com/api/v1/";
+
+  const CATEGORIES_PATH = "categories",
+    PLACES_PATH = "places/",
+    USER_PATH = "user/",
+    LOGIN_PATH = "user/sign_in/",
+    REGISTER_PATH = "user/sign_up/",
+    REVIEWS_PATH = "reviews/",
+    CREATE_REVIEW_PATH = "/reviews#create",
+    CREATE_PLACE_PATH = "places#create";
```
Comment (dwoznicka) [inline app/resources/js/api-client.js]: why does one path start with a slash and the other is not?

File: app/resources/js/index.js [hunk 8]
```diff
@@ -0,0 +1,165 @@
+const app = (function () {
+
+  const logError = function(error) {
+    console.log(error);
```
Comment (dwoznicka) [inline app/resources/js/index.js]: not really a good error handling :) besides logging (in real world we would log that to server ) we should display some information for the user like `something went wrong, try again`. as a popup or just error label

File: app/resources/js/api-client.js (hunk 7 above)
Comment (MaciejOstojski4) [reply app/resources/js/api-client.js]: It depends on where it's located in url. If it is after API_URL path, then it hasn't slash, but if it's in the middle than it has one.

File: app/resources/sass/components/placeCard.scss (hunk 6 above)
Comment (MaciejOstojski4) [reply app/resources/sass/components/placeCard.scss]: I think, I can't add custom css style to that element (it's from jquery barrating plugin), so I must use such long css selector to add some style to it.
